services/QuillLMS/client/app/bundles/Staff/components/activityForm/topicGroup.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react'
+
+import TopicColumn from './topicColumn'
+
+const TopicGroup = ({ createNewTopic, findParentTopic, topic, topicOptions, handleLevelOneChange }) => {
+
+  const [levelOneTopic, setLevelOneTopic] = React.useState(topic)
+  const [levelTwoTopic, setLevelTwoTopic] = levelOneTopic.id ? React.useState(findParentTopic(topic)) : React.useState({})
+  const [levelThreeTopic, setLevelThreeTopic] = levelOneTopic.id ? React.useState(findParentTopic(levelTwoTopic)) : React.useState({})
+
+  React.useEffect(() => {
+    if (levelOneTopic.parent_id !== levelTwoTopic.id) {
+      handleLevelOneChange(levelOneTopic.id, null)
```

Comment:
It feels like a little bit of a code smell to explicitly call `handleLevelOneChange` before you set it when we could use an effect hook to do that when the value changes. That's a little complicated by the fact that you need the old value to pass up to the function, but you could do something like [this](https://stackoverflow.com/questions/53446020/how-to-compare-oldvalues-and-newvalues-on-react-hooks-useeffect) which I think would still be cleaner. Up to you, just wanted to flag it as a little unparadigmatic.